### PR TITLE
Swift 3 Beta 6 Support

### PR DESCRIPTION
**Pull request by Legoless**
This PR contains initial work done for the project to work with Xcode 8 Beta 6. Needs to be tested, before merged.

### CODE REVIEW

File: Eureka.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -396,6 +396,7 @@
 				TargetAttributes = {
 					51729DEA1B9A4F5E004A00EB = {
 						CreatedOnToolsVersion = 7.0;
+						DevelopmentTeam = 289M6XEDV4;
```
Comment (mtnbarreto) [inline Eureka.xcodeproj/project.pbxproj]: @Legoless  Could we remove this attribute from target configuration?

File: Eureka.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -647,6 +650,7 @@
 				APPLICATION_EXTENSION_API_ONLY = YES;
 				CLANG_ENABLE_MODULES = YES;
 				DEFINES_MODULE = YES;
+				DEVELOPMENT_TEAM = 289M6XEDV4;
```
Comment (mtnbarreto) [inline Eureka.xcodeproj/project.pbxproj]: same here.

File: Example.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -491,6 +492,7 @@
 			buildSettings = {
 				ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES = YES;
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
+				DEVELOPMENT_TEAM = 289M6XEDV4;
```
Comment (mtnbarreto) [inline Example.xcodeproj/project.pbxproj]: here too

File: Example.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -506,6 +508,7 @@
...
```
Comment (mtnbarreto) [inline Example.xcodeproj/project.pbxproj]: ;)

**Comment by Legoless** [comment]
Oh, yeah, of course. This stayed in because otherwise it won't compile.

**Comment by Legoless** [comment]
Fixed and squashed.

File: Example/Example/CustomCells.swift [hunk 5]
```diff
@@ -152,14 +152,14 @@ public class _FloatLabelCell<T where T: Equatable, T: InputTypeInitiable>: Cell<
     lazy public var floatLabelTextField: FloatLabelTextField = { [unowned self] in
         let floatTextField = FloatLabelTextField()
         floatTextField.translatesAutoresizingMaskIntoConstraints = false
-        floatTextField.font = .preferredFont(forTextStyle: UIFontTextStyleBody)
+        floatTextField.font = .preferredFont(forTextStyle: UIFontTextStyle.body)
```
Comment (mtnbarreto) [inline Example/Example/CustomCells.swift]: Could we use just `.body` ?

File: Example/Example/Info.plist [hunk 6]
```diff
@@ -35,6 +35,7 @@
 		<string>UIInterfaceOrientationPortrait</string>
 		<string>UIInterfaceOrientationLandscapeLeft</string>
 		<string>UIInterfaceOrientationLandscapeRight</string>
+		<string>UIInterfaceOrientationPortraitUpsideDown</string>
```
Comment (mtnbarreto) [inline Example/Example/Info.plist]: please remove this line.

File: Example/Example/ViewController.swift [hunk 7]
```diff
@@ -429,26 +429,26 @@ class FieldRowCustomizationController : FormViewController {
...
                 }
                 <<< NameRow() {
                     $0.title = "Another Title"
                     $0.textFieldPercentage = 0.6
                     $0.placeholder = "textFieldPercentage = 0.6"
                 }
                 .cellUpdate {
-                    $0.cell.textField.textAlignment = .left
-                    $0.cell.textLabel?.textAlignment = .right
+                    $1.cell.textField.textAlignment = .left
```
Comment (mtnbarreto) [inline Example/Example/ViewController.swift]: I think we can use `$1.textField ` which is shorter.

File: Source/Core/Core.swift [hunk 8]
```diff
@@ -197,7 +197,7 @@ public enum PresentationMode<VCType: UIViewController> {
                     fatalError()
                 }
                 porpoverController.sourceView = porpoverController.sourceView ?? presentingViewController.tableView
-                porpoverController.sourceRect = porpoverController.sourceRect ?? row.baseCell.frame
+                //porpoverController.sourceRect = porpoverController.sourceRect ?? row.baseCell.frame
```
Comment (mtnbarreto) [inline Source/Core/Core.swift]: @Legoless  Why dod you comment this line?

Comment (mats-claassen) [inline Source/Core/Core.swift]: Why did you comment this line?

File: Source/Core/BaseRow.swift [hunk 9]
```diff
@@ -114,8 +114,8 @@ public class BaseRow : BaseRowType {
         return IndexPath(row: rowIndex, section: sectionIndex)
     }
     
-    private var hiddenCache = false
-    private var disabledCache = false {
+    var hiddenCache = false
```
Comment (mtnbarreto) [inline Source/Core/BaseRow.swift]: @Legoless Is this an automatic change by Xcode? we should use `fileprivate` (old private) or `private`.

**Comment by mtnbarreto** [comment]
@Legoless  great work!! Let's merge this, we can make another pull request to fix some minor issues.

**Pull request merged**